src/test_report.py

```diff
@@ -112,47 +109,50 @@ def _get_report(self, root_node):
         return content, subject
 
     def _send_report(self, subject, content):
-        try:
-            self._email_sender.create_and_send_email(
-                    subject, content
-                )
-        except Exception as err:
-            self._logger.log_message(logging.ERROR, err)
-
-    def loop(self):
-        """Method to execute in a loop"""
-        sub_id = self._db.subscribe_node_channel(filters={
+        self._email_sender.create_and_send_email(subject, content)
```

Comment:
`try-except` block shouldn't be dropped from here. As this can result in potential exceptions from SMTP lib.
Or else, it should be handled in `email_sender` module and a resultant status can be received here.
```suggestion
        try:
            self._email_sender.create_and_send_email(
                    subject, content
                )
        except Exception as err:
            self.log.error(err)
```

Reply:
I see this is already been handled in `Service.run` method.

Reply:
Exactly, which is why it can be dropped here.